src/pydrex/interpolations.py

```diff
@@ -0,0 +1,281 @@
+                    it.chain(
+                        *map(
+                            create_interpolators,
+                            (
+                                "CloughTocher2DInterpolator",
+                                "CloughTocher2DInterpolator",
+                                "NearestNDInterpolator",
+                            ),
+                            [coords] * 3,
+                            (velocity, velocity_gradient, deformation_mechanism),
+                        )
+                    ),
+                )
+            )
+
+    else:  # 3D
+        interpolators = dict(
+            zip(
+                [
+                    "velocity_x",
+                    "velocity_y",
+                    "velocity_z",
+                    "grad_velocity_xx",
+                    "grad_velocity_yx",
+                    "grad_velocity_zx",
+                    "grad_velocity_xy",
+                    "grad_velocity_yy",
+                    "grad_velocity_zy",
+                    "grad_velocity_xz",
+                    "grad_velocity_yz",
+                    "grad_velocity_zz",
+                    "deformation_mechanism",
+                ],
+                it.chain(
+                    *map(
+                        create_interpolators,
+                        ["NearestNDInterpolator"] * 3,
+                        [coords] * 3,
+                        (velocity, velocity_gradient, deformation_mechanism),
+                    )
+                ),
+            )
+        )
+
+    return interpolators
+
+
+def create_interpolators(interpolator, coords, data, triangles=None, **kwargs):
+    """Create interpolator callbacks for data arrays.
+
+    Args:
+        `intepolator` (string) — name of interpolator class to use, see below
+        `coords` (2D or 3D NumPy array) — coordinates of the finite element mesh
+        `data` (NumPy array) — data used to create the interpolators
+        `triangles` (optional) — 2D triangle vertex indices, see below
+
+    Optional kwargs will be passed to the interpolation constructor.
+
+    Returns a tuple of interpolator callbacks, one for each data vector component.
+    For scalar data, the tuple contains only one interpolator.
+
+    The `triangles` arg is required only for "CubicTriInterpolator".
+    See the documentation of that constructor for details.
+
+    Supported interpolator choices (dim = `coords.shape[1]`):
+
+    |                                 |   scalar data   |   vector data   |
+    |       interpolator              |    and dim ==   |    and dim ==   |
+    |                                 |   2    |   3    |   2    |   3    |
+    |---------------------------------------------------------------------|
+    | "CloughTocher2DInterpolator" [1]|        |        |   x    |        |
+    | "CubicTriInterpolator" [2]      |        |        |   x    |        |
+    | "NearestNDInterpolator" [3]     |   x    |   x    |        |   x    |
```

Comment:
Why do you distinguish between scalar and vector? Those interpolators only work for scalar data, or am I missing something?

Reply:
Ah, it's not very clear. By "scalar" I meant a vector of scalars, see https://github.com/Patol75/PyDRex/pull/9/files/b29d96fac5159cacfb8bb0c9e0980af5349bc8dd#diff-4281486f0b06da1f6a2e3a09a74550f908b703735f2974f31632131bc0f81806R207

What I'm trying to do is always use NearestNDInterpolator for things like the DeformationMechanism field, equivalent to https://github.com/Patol75/PyDRex/blob/master/src/PyDRex.py#L773

Reply:
Maybe there is a better interpolator class to use here.

Reply:
Looks like deformation mechanism always has shape `(n,1)` where `n` is the number of nodes:

```
In [5]: defmech = _vtk.read_tuple_array(pointdata, "DeformationMechanism")

In [6]: defmech
Out[6]:
array([[1.],
       [1.],
       [1.],
       ...,
       [1.],
       [1.],
       [1.]])

In [7]: defmech[0]
Out[7]: array([1.])

In [8]: defmech.shape
Out[8]: (1584, 1)
```

Maybe it's best to modify the `is_scalar` condition to not take arrays of shape `(n,)`.

Reply:
Wait, this does not have anything to do with shape. The deformation mechanism is a discrete field, so you need to do an interpolation based on the nearest neighbour algorithm, otherwise, you end up with meaningless values. For continuous fields, I use the Clough-Tocher scheme in 2-D because it is fast enough. In 3-D, I fall back to the nearest neighour because the linear interpolation scheme is usually too slow.

Reply:
OK I guess I was assuming that the only scalar field input would be deformation mechanism. I'll remove the confusing validator and change the docs.